models/src/main/java/com/vimeo/networking2/adapters/CustomDateAdapter.kt

```diff
@@ -0,0 +1,45 @@
+package com.vimeo.networking2.adapters
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import java.io.IOException
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * Custom adapter that returns a null value if the date
+ * was not specified in the response. Otherwise, it will
+ * attempt to parse the date.
+ */
+class CustomDateAdapter : JsonAdapter<Date>() {
+
+    @Synchronized
+    @Throws(IOException::class)
+    override fun fromJson(reader: JsonReader): Date? {
+        val token: JsonReader.Token = reader.peek()
+        return if (token == JsonReader.Token.NULL) {
+            reader.skipValue()
+            null
+        } else {
+            parseDate(reader)
```

Comment:
I know there was concern about needing to write this class due to the null safety. I looked at the source of Moshi and it's possible that we won't need this class, assuming that the date formats supported by moshi are the ones our API supports.

Since this is the normal JSON Date adapter used by Moshi: https://github.com/square/moshi/blob/master/adapters/src/main/java/com/squareup/moshi/Rfc3339DateJsonAdapter.java

We can just do a `nullSafe` call on an instance of the adapter and we will be able to use it without having to write a custom adapter. See the function here: https://github.com/square/moshi/blob/master/moshi/src/main/java/com/squareup/moshi/JsonAdapter.java#L130

How it will be consumed when setting up moshi:
```kotlin
.add(Date::class.java, Rfc3339DateJsonAdapter().nullSafe())
```
If Moshi is using a different date format then this won't work I don't think.

Reply:
You're right. I updated the initialization to use `nullSafe`.